ssm/variational.py

```diff
@@ -256,23 +256,74 @@ def _initialize_variational_params(self, data, input, mask, tag):
         # mu = J^{-1} h and Sigma = J^{-1}.  Initialize J to inverse of
         # initial variance and scale h accordingly, so the mean is the output
         # of the emissions invert function.
-        J_diag = np.tile(1.0 / self.initial_variance * np.eye(D)[None, :, :], (T, 1, 1))
-        J_lower_diag = np.zeros((T-1, D, D))
+        # J_diag = np.tile(1.0 / self.initial_variance * np.eye(D)[None, :, :], (T, 1, 1))
+        # J_lower_diag = np.zeros((T-1, D, D))
+        # if self.model.emissions.single_subspace:
+        #     h = (1.0 / self.initial_variance) \
+        #         * self.model.emissions.invert(data, input=input, mask=mask, tag=tag)
+        # else:
+        #     # TODO smarter inversion with multiple subspace!
+        #     warn("Posterior initialization is not implemented for multiple subspaces. \
+        #           A random initialization is used.")
+        #     h = (1.0 / self.initial_variance) * npr.randn(data.shape[0], self.D)
+
+        # Initialize q(x) = N(J, h) where J is a block tridiagonal precision matrix
+        # and h is the linear potential.  The mapping to mean parameters is
+        # mu = J^{-1} h and Sigma = J^{-1}.  We will represent J and h in terms of
+        #
+        # Initial distribution parameters:
+        # J_ini:     (D, D)       initial state precision
+        # h_ini:     (D,)         initial state bias
+
+        # If time-varying dynamics:
+        # J_dyn_11:  (T-1, D, D)  upper left block of dynamics precision
+        # J_dyn_21:  (T-1, D, D)  lower left block of dynamics precision
+        # J_dyn_22:  (T-1, D, D)  lower right block of dynamics precision
+        # h_dyn_1:   (T-1, D)     upper block of dynamics bias
+        # h_dyn_2:   (T-1, D)     lower block of dynamics bias
+        #
+        # Observation distribution parameters
+        # J_obs:     (T, D, D)    observation precision
+        # h_obs:     (T, D)       observation bias
+        #
+        # Initialize the posterior to mean zero
+        h_ini = np.zeros(D)
+        h_dyn_1 = np.zeros((T-1, D))
+        h_dyn_2 = np.zeros((T-1, D))
+        h_obs = np.zeros((T, D))
+
         if self.model.emissions.single_subspace:
-            h = (1.0 / self.initial_variance) \
-                * self.model.emissions.invert(data, input=input, mask=mask, tag=tag)
-        else:
-            # TODO smarter inversion with multiple subspace!
-            warn("Posterior initialization is not implemented for multiple subspaces. \
-                  A random initialization is used.")
-            h = (1.0 / self.initial_variance) * npr.randn(data.shape[0], self.D)
+            h_obs = self.model.emissions.\
```

Comment:
Let's bring back the `1 / self.initial_variance` multiplier to `h_obs` and `J_obs`.